### PR TITLE
Add face_grad_op for 3d topologies

### PR DESCRIPTION
**Pull request by rcknaus**
* Hex8, Hex27, Tet4, Wed6, and Pyr5 with a unit test comparing to the old ME values 
* Renames some of the algtraits.  I'm not particular about it; it's just what I had originally
*  Switches back from faceOrdinals to faceOrdinal in scratchviews, given that we now have a uniform collection of face ordinals over the simd workset

### CODE REVIEW

File: src/master_element/Pyr5CVFEM.C [hunk 1]
```diff
@@ -841,6 +842,64 @@ void PyrSCS::face_grad_op(
         std::cout << "problem with PyrSCS::face_grad_op." << std::endl;
     }
   }
+
+  for (int j = 0; j < 15; ++j) {
+    std::cout << gradop[j] << std::endl;
```
Comment (alanw0) [inline src/master_element/Pyr5CVFEM.C]: This cout statement probably should be deleted.

File: src/master_element/Pyr5CVFEM.C [hunk 2]
```diff
@@ -841,6 +842,64 @@ void PyrSCS::face_grad_op(
         std::cout << "problem with PyrSCS::face_grad_op." << std::endl;
```
Comment (sayerhs) [inline src/master_element/Pyr5CVFEM.C]: Will this be output even during parallel runs?

File: include/master_element/Pyr5CVFEM.h [hunk 3]
```diff
@@ -173,6 +176,11 @@ class PyrSCS : public MasterElement
     double *det_j,
     double *error);
 
+  void face_grad_op(
+    int face_ordinal,
+    SharedMemView<DoubleType**>& coords,
+    SharedMemView<DoubleType***>& gradop) final;
```
Comment (spdomin) [inline include/master_element/Pyr5CVFEM.h]: Why the usage of final - just curious.

File: include/AlgTraits.h [hunk 4]
```diff
@@ -117,6 +117,153 @@ struct AlgTraitsHexGL {
   static constexpr stk::topology::topology_t baseTopo_ = stk::topology::HEX_8;
 };
 
+struct AlgTraitsEdge_3D
+{
+  static constexpr int nDim_ = 3;
+  static constexpr int nodesPerElement_ = 3;
+  static constexpr int numScsIp_ = 2;
+  static constexpr int numScvIp_ = 0;
```
Comment (spdomin) [inline include/AlgTraits.h]: I am not sure things like, i.e., numScvIp_ are required? The only place such variables are used are in the kernels, correct?

File: src/pmr/RadTransWallElemKernel.C [hunk 5]
```diff
@@ -31,15 +31,15 @@ RadTransWallElemKernel<BcAlgTraits>::RadTransWallElemKernel(
   ElemDataRequests &dataPreReqs)
   : Kernel(),
     radEqSystem_(radEqSystem),
-    ipNodeMap_(sierra::nalu::MasterElementRepo::get_surface_master_element(BcAlgTraits::faceTopo_)->ipNodeMap())
+    ipNodeMap_(sierra::nalu::MasterElementRepo::get_surface_master_element(BcAlgTraits::topo_)->ipNodeMap())
```
Comment (spdomin) [inline src/pmr/RadTransWallElemKernel.C]: Why this change? I can not seem to see any changes in BcAlgTraits despite a number of new AlgTraits.

File: include/AlgTraits.h [hunk 6]
```diff
@@ -117,6 +117,153 @@ struct AlgTraitsHexGL {
   static constexpr stk::topology::topology_t baseTopo_ = stk::topology::HEX_8;
 };
 
+struct AlgTraitsEdge_3D
+{
+  static constexpr int nDim_ = 3;
+  static constexpr int nodesPerElement_ = 3;
```
Comment (spdomin) [inline include/AlgTraits.h]: Is this correct? nodesPerElement 3 with scsIp = 2?

File: include/AlgTraits.h [hunk 7]
```diff
@@ -120,10 +120,9 @@ struct AlgTraitsHexGL {
 struct AlgTraitsEdge_3D
 {
   static constexpr int nDim_ = 3;
-  static constexpr int nodesPerElement_ = 3;
-  static constexpr int numScsIp_ = 2;
-  static constexpr int numScvIp_ = 0;
-  static constexpr int numGp_ = 2; // for FEM
+  static constexpr int nodesPerElement_ = 2;
+  static constexpr int numScsIp_ = 1;
+  static constexpr int numScvIp_ = 2;
```
Comment (spdomin) [inline include/AlgTraits.h]: I do not think that numScvIp_ is required, however, you need not remove this prior to push.

**Comment by spdomin** [comment]
@rcknaus, I am still unclear as to why this original code change caused so many diffs/fails - especially since the expose face_grad_op is not even activated in the tests that were failing.

**Pull request merged**